code/controllers/subsystem/ticker.dm

```diff
@@ -394,7 +394,7 @@ SUBSYSTEM_DEF(ticker)
 			continue
 		var/datum/job/player_assigned_role = new_player_living.mind.assigned_role
 		if(ishuman(new_player_living) && CONFIG_GET(flag/roundstart_traits))
-			if(new_player_mob.client.prefs.should_be_random_hardcore(player_assigned_role, new_player_living.mind))
+			if(new_player_mob.client?.prefs?.should_be_random_hardcore(player_assigned_role, new_player_living.mind))
 				new_player_mob.client.prefs.hardcore_random_setup(new_player_living)
 			SSquirks.AssignQuirks(new_player_living, new_player_mob.client)
```

Comment:
Do we need to be concerned about this `.client` and the one on line 401?

Reply:
this will only run if there is a client, wont it

Reply:
The client parameter is optional for `EquipRank()`
The clientless spawn will simply not have its prefs respected.